File: Zad2/BottomNavigation/app/src/main/java/pl/pjwstk/bottomnavigation/ui/home/HomeFragment.java
package pl.pjwstk.bottomnavigation.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import pl.pjwstk.bottomnavigation.R;
import pl.pjwstk.bottomnavigation.databinding.FragmentHomeBinding;

public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;

    private EditText ETPersonWeight;
    private EditText ETPersonHeight;
    private Button BCalculate;
    private TextView BMIInfo;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        HomeViewModel homeViewModel =
                new ViewModelProvider(this).get(HomeViewModel.class);

        binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        ETPersonWeight = root.findViewById(R.id.editTextTextPersonWeight);
        ETPersonHeight = root.findViewById(R.id.editTextTextPersonHeight);
        BMIInfo = root.findViewById(R.id.textViewBMIInfo);
        BCalculate = root.findViewById(R.id.buttonCalculate);

        BCalculate.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                CalCulateBMI();
            }
        });

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    private void CalCulateBMI() {
        float weight = Float.parseFloat(ETPersonWeight.getText().toString());
        float height = Float.parseFloat(ETPersonHeight.getText().toString());

        float bmi = (100*100*weight)/(height*height);

        BMIInfo.setText("Your BMI is: " + String.format("%.2f", bmi));
    }
}

File: Zad4/zadanie4/app/src/main/java/pl/pjwstk/zadanie4/ui/home/HomeFragment.kt
package pl.pjwstk.zadanie4.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import pl.pjwstk.zadanie4.R
import pl.pjwstk.zadanie4.databinding.FragmentHomeBinding

class HomeFragment : Fragment() {

    private var _binding: FragmentHomeBinding? = null
    private var ETPersonWeight: EditText? = null
    private var ETPersonHeight: EditText? = null
    private var BCalculate: Button? = null
    private var BMIInfo: TextView? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val homeViewModel =
            ViewModelProvider(this).get(HomeViewModel::class.java)

        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        val root: View = binding.root

        ETPersonWeight = root.findViewById<EditText>(R.id.editTextTextPersonWeight)
        ETPersonHeight = root.findViewById<EditText>(R.id.editTextTextPersonHeight)
        BMIInfo = root.findViewById<TextView>(R.id.textViewBMIInfo)
        BCalculate = root.findViewById<Button>(R.id.buttonCalculate)

        BCalculate!!.setOnClickListener(View.OnClickListener { CalCulateBMI() })

        return root
    }

    private fun CalCulateBMI() {
        val weight: Float = ETPersonWeight!!.getText().toString().toFloat()
        val height: Float = ETPersonHeight!!.getText().toString().toFloat()
        val bmi = 100 * 100 * weight / (height * height)
        BMIInfo!!.setText("Your BMI is: " + String.format("%.2f", bmi))
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}